pkg/arvo/app/publish.hoon

```diff
@@ -242,8 +244,100 @@
       ==
     ::
         %6
+      %_  $
+        -.p.old-state  %7
+        ::
+          cards
+        %+  weld  cards
+        %+  roll  ~(tap by books.p.old-state)
+        |=  [[[who=@p book=@tas] nb=notebook] out=(list card)]
+        ^-  (list card)
+        =/  =resource
+          [who book]
+        ?.  =(who our.bol)
+          :_  out
+          (poke-graph-pull %add who resource)
+        =/  =graph:graph-store
+          (notebook-to-graph nb)
+        %+  weld  out
+        ^-  (list card)
+        :~ 
+          %-  poke-graph-store
+          :*  %0  date-created.nb  %add-graph
+              resource 
+              graph
+              `%graph-validator-publish
+          ==
+          (poke-graph-push %add resource)
+        ==
+      ==
+    ::
+        %7
       [cards this(state p.old-state)]
     ==
+    ++  notebook-to-graph
+      |=  =notebook
+      ^-  graph:graph-store
+      %+  gas:orm:graph-store  *graph:graph-store
+      %+  turn  ~(tap by notes.notebook)
+      |=  [@ta =note]
+      ^-  [atom node:graph-store]
+      :-  date-created.note
+      (note-to-node note)
+    ::
+    ++  comments-to-children
+      |=  [=index:graph-store =note]
+      ^-  internal-graph:graph-store
+      ?:  =(~ comments.note)
+        [%empty ~]
+      :-  %graph
+      %+  gas:orm:graph-store  *graph:graph-store
+      %+  turn  ~(tap by comments.note)
+      |=  [when=@da =comment]
+      ^-  [atom node:graph-store]
+      :-  when
+      :_  [%empty ~]
+      ^-  post:graph-store
+      :*  author.comment
+          (snoc index when)
+          when
+          [%text content.comment]~
+          ~  ~
+      ==
+    ::
+    ++  note-to-node
+      |=  =note
+      ^-  node:graph-store
+      =/  =index:graph-store
+        ~[date-created.note]
+      =/  contents=(list content:graph-store)
+        :~  [%text title.note]
+            [%text file.note]
+        ==
+      :_  (comments-to-children index note)
```

Comment:
We need the `%1` child for revisions and the `%2` pattern for comments in the publish mark to be reflected here. It is not currently. The top-level node should have null contents and the `%1` and `%2` child that is in the mark.